src/index.rs

```diff
@@ -293,3 +407,141 @@ fn index_single_block(
         .expect("in-memory writers don't error");
     debug_assert_eq!(len, BlockHash::LEN);
 }
+
+fn scan_single_block_for_silent_payments(
+    index: &Index,
+    daemon: &Daemon,
+    block_hash: BlockHash,
+    block: SerBlock,
+    batch: &mut WriteBatch,
+) {
+    struct IndexBlockVisitor<'a> {
+        daemon: &'a Daemon,
+        index: &'a Index,
+        map: &'a mut HashMap<BlockHash, Vec<u8>>,
+    }
+
+    impl<'a> Visitor for IndexBlockVisitor<'a> {
+        fn visit_transaction(&mut self, tx: &bsl::Transaction) -> core::ops::ControlFlow<()> {
+            let parsed_tx: bitcoin::Transaction = match deserialize(tx.as_ref()) {
+                Ok(tx) => tx,
+                Err(_) => panic!("Unexpected invalid transaction"),
+            };
+
+            if parsed_tx.is_coinbase() { return ControlFlow::Continue(()) };
+
+            let txid = bsl_txid(tx);
+
+            let mut to_scan = false;
+            for (i, o) in parsed_tx.output.iter().enumerate() {
+                if o.script_pubkey.is_p2tr() {
+                    let outpoint = OutPoint {
+                        txid,
+                        vout: i.try_into().expect("Unexpectedly high vout"),
+                    };
+                    if self
+                        .index
+                        .store
+                        .iter_spending(SpendingPrefixRow::scan_prefix(outpoint))
+                        .next()
+                        .is_none()
```

Comment:
If I understand correctly, this must make sure only transactions with (P2TR) unspent outputs are scanned.
Making an index dependent on data from another index seems odd to me, I have some questions about this:
- Does this handle collisions of `spending` entries in the database (correctly)?
- Wouldn't a user want historic data after restoring a wallet?
- Looking at the code, `spending` is made sure to be updated to the current chain tip and then `tweaks` is indexed (do I get this correctly?). This will include spent transactions (imagine 1 P2TR output TXs) as blocks get propagated, but not on the initial sync or when spent in the same block. Is this intended?